libcaf_core/test/config_option_set.cpp

```diff
@@ -44,8 +44,9 @@ CAF_TEST_FIXTURE_SCOPE(config_option_set_tests, fixture)
 
 CAF_TEST(lookup) {
   opts.add<int>("opt1,1", "test option 1")
-    .add<float>("test", "opt2,2", "test option 2")
-    .add<bool>("test", "flag,fl3", "test flag");
+    .add<float>("test", "opt2,2", "test option 2");
+  auto flag = make_config_option<bool>("test", "flag,fl3", "test flag");
```

Comment:
I would've preferred new unit tests for the copying. This mixes copy-semantic testing into the lookup test, making this test harder to read and reason about.